packages/vm/src/wasm_backend/store.rs

```diff
@@ -18,7 +18,7 @@ use super::limiting_tunables::LimitingTunables;
 /// is 65536 (2^16) bytes. In WebAssembly version 1, a linear memory can have at
 /// most 65536 pages, for a total of 2^32 bytes (4 gibibytes).
 /// https://github.com/WebAssembly/memory64/blob/master/proposals/memory64/Overview.md
-const MAX_WASM_MEMORY: usize = 4 * 1024 * 1024 * 1024;
+const MAX_WASM_MEMORY: usize = u32::MAX as usize;
```

Comment:
> [WebAssembly linear memory objects](https://webassembly.github.io/spec/core/syntax/modules.html#memories) have sizes measured in [pages](https://webassembly.github.io/spec/core/exec/runtime.html#page-size). Each page is 65536 (2^16) bytes. In WebAssembly version 1, a linear memory can have at most 65536 pages, for a total of 2^32 bytes (4 [gibibytes](https://en.wikipedia.org/wiki/Gibibyte)).

The value should be 4294967296 (4 GiB), not 4294967295 (2**32-1). Just the largest memory location is 4294967295 because counting starts at 0. That's the same as 0, 1, 2, 3 is 4 elements.

But the code can be rewritten in a way to overcome the limitation that you cannot store 4294967296 in an `usize` on a 32 bit system. Will send a PR.

Reply:
https://github.com/CosmWasm/cosmwasm/pull/1346

Reply:
Could you rebase this work into current main? The issue should be solved there and d2d54891179db1bcd65813883709dc3fe303c716 can be skipped.